src/docs/testing/errors.md

```diff
@@ -84,9 +90,47 @@ class MyApp extends StatelessWidget {
 }
 ```
 
+## Set up error handling for a Zone
+
+Consider an `onPressed` callback that invokes an asynchronous function,
+such as `MethodChannel.invokeMethod` (or pretty much any plugin).
+For example:
+<!-- skip -->
+```dart
+OutlinedButton(
+  child: Text('Click me!'),
+  onPressed: () async {
+    final channel = const MethodChannel('crashy-custom-channel');
+    await channel.invokeMethod('blah');
+  },
+),
+```
+If `invokeMethod` throws an error, it won't be forwarded to `FlutterError.onError`.
+Instead, it's forwarded to the `Zone` where `runApp` was run.
+
+To catch such an error, use [`runZonedGuarded`][].
+<!-- skip -->
+```dart
+import 'dart:async';
+
+void main() {
+  runZonedGuarded(() async {
+    WidgetsFlutterBinding.ensureInitialized();
+    runApp(MyApp()));
+  }, (Object error, StackTrace stack) {
```

Comment:
Is `WidgetsFlutterBinding.ensureInitialized()` really necessary here? If I'm not mistaken, it is also called from `runApp()`. I don't see why it needs to be called twice in a row in the same zone.

Reply:
It is indeed not necessary it this example.

But in many cases `WidgetsFlutterBinding.ensureInitialized()` needs to be called manually if a developer wants to initialize something before `runApp` (e.g. Crashlytics).

`WidgetsFlutterBinding.ensureInitialized()` is called manually in this example to emphasize that it needs to be called from inside of the `Zone`, otherwise errors handling won't work.

Reply:
@blazern
Yes, but does it have to be inside `runZonedGuarded()` even if the initialization you mentioned is outside of it, or do you mean all initialization should be done inside `runZonedGuarded()` like below?

```dart
void main() {
  runZonedGuarded(() async {
    WidgetsFlutterBinding.ensureInitialized();
    await Firebase.initializeApp();
    runApp(MyApp());
  }, ...);
}
```

Reply:
@kaboc
I mean that if `WidgetsFlutterBinding.ensureInitialized()` is to be called, it must always be called inside `runZonedGuarded()`.

And if some initialization (like `await Firebase.initializeApp()`) must be called after `WidgetsFlutterBinding.ensureInitialized()`, then such initialization must also be called from `runZonedGuarded()`.

I thought that it can be easily understood from [the note below](https://github.com/flutter/website/pull/5612/files#diff-913f1e4f31d4c1700b06e4c3c4653f4470fe6c8f088b258fac6143a944767d3aR125)... Do you think the note should be reworded?

Reply:
@blazern 
Now that I understand what you meant, the note looks easy to understand, but it was confusing at the first glance because I'd never seen pieces of code where initialization was done inside `runZonedGuarded()`.

How about adding some comment about initialization between `WidgetsFlutterBinding.ensureInitialized();` and `runApp(MyApp());` in the example code? I guess it'll help readers understand it more clearly.